packages/turndown/src/commonmark-rules.js

```diff
@@ -140,6 +140,43 @@ rules.foregroundColor = {
   },
 }
 
+// Converts placeholders for not-loaded resources.
+rules.resourcePlaceholder = {
+  filter: function (node, options) {
+    if (!options.allowResourcePlaceholders) return false;
+    if (!node.classList || !node.classList.contains('not-loaded-resource')) return false;
+    const isImage = node.classList.contains('not-loaded-image-resource');
+    if (!isImage) return false;
+
+    const resourceId = node.getAttribute('data-resource-id');
+    return resourceId && resourceId.match(/^[a-z0-9]{32}$/);
+  },
+
+  replacement: function (_content, node) {
+    const htmlBefore = node.getAttribute('data-original-before') || '';
+    const htmlAfter = node.getAttribute('data-original-after') || '';
+    const isHtml = htmlBefore || htmlAfter;
+    const resourceId = node.getAttribute('data-resource-id');
+    if (isHtml) {
+      const attrs = [
+        htmlBefore.trim(),
+        `src=":/${resourceId}"`,
+        htmlAfter.trim(),
+      ].filter(a => !!a);
+
+      return `<img ${attrs.join(' ')}>`;
```

Comment:
Look in to a way to reduce code duplication: Pass a function through `options` to do part of the conversion? Alternatively, add to `utils`.